### DISCUSSION TITLE
Grid 12, 16 e 24 quando usar?

### DISCUSSION THREAD

**Discussion by Marcelosilva10** [frontendbr/forum#154] [Discussão]
Pessoal em qual momento é ideal para ser usado um grid de 12, 16 ou grid 24?

**Comment by felipefialho** [comment]
Você está perguntando sobre o Layout ou sobre o código?

Se for o código, acredito que o grid deve ser definido junto como Designer, para que a maior fidelidade possível seja alcançada.

Caso o layout não tenha grid estabelecido, cabe a você tentar dividir as colunas para chegar em um padrão.

Outra coisa, apesar de "12, 16 e 24" serem grids bastante utilizados, não são o únicos. Não é incomum usarmos grids de 6 ou 8 nos projetos aqui na empresa.

**Comment by RicardoJunior** [comment]
Eu prefiro quando o designer e o dev acertam o grid juntos, pra ficar algo mais "pixel perfect", e eu acho que o grid é váriavel, depende do projeto. Quem faz esses paranauês e define qual grid iremos usar desde os desenhos é o @alexandrehappybiz, ele pode falar alguma coisa sobre isso também.

**Comment by abnersajr** [comment]
@Marcelosilva10 Geralmente se usa esses números por questões de proporções, golden ratio e umas frescurite de designer :yum: mas na verdade varia de cada projeto. 
Vamos dar um exemplo:
Bootstrap tem 12 colunas, mas em seu projeto a maioria das coisas são desenhadas usando divisões de 3 em 3 colunas logo você poderia ao inves de ter um grid de 1 até 12 de classes um de 1 até 4
1-4 e 3-12 fariam o mesmo efeito.

**Comment by filipelinhares** [comment]
Não vejo porque tantas colunas, nunca precisei usar mais que 12. Concordo com o @LFeh que você deve sentar com o designer e decidirem isso juntos pra ficar melhor pra todo mundo.

Tenho usado grides por fração ultimamente e tenho achado mais intuitivo e simples.

Exemplos:
- [Primercss](https://github.com/primer/primer/blob/master/scss/_layout.scss#L28-L54)
- [Um projeto meu que uso](https://github.com/filipelinhares/funkcss-components/blob/master/lib/grid.css#L54-L60)
- [Purecss grid](http://purecss.io/grids/)
